### ISSUE TITLE
Reproduce result from Co-Fusion paper

### ISSUE THREAD

**Issue by ung264** [christian-rauch/MultiMotionFusion#3]
Hi,

Thanks for sharing the great work!

I was trying to reproduce the result of Co-Fusion and came across your repo since it supports ubuntu 20.04. I managed to build and run MMF. But when I tried to evaluate MMF on car4 and room4 sequence, I couldn't get the same result as in Co-Fusion paper, after disabling the extra command line arguments from MMF. Do you have any experience on how to reproduce CF paper result for car4 and room4 sequences? If so, could you maybe share your the arguments used?  Thanks in advance.

**Comment by christian-rauch** [comment]
The `MultiMotionFusion` executable is supposed to behave like the Co-Fusion baseline in the default configuration without the proposed extensions. However, I've not tried to reproduce the results on the original `klg` log files.

I applied a couple of fixes on top of the upstream Co-Fusion to fix compiler warnings and errors. Those fixes can be found in the PR https://github.com/martinruenz/co-fusion/pull/20, which is the same as the [`fix_warnings` branch](https://github.com/christian-rauch/MultiMotionFusion/tree/fix_warnings) in this repo. This PR/branch is sufficient, if you just want to run on the `klg` log format.

The interface to ROS and additional fixes are part of the [`rosnode` branch](https://github.com/christian-rauch/MultiMotionFusion/tree/rosnode). Some of those fixes were required to run on GPUs with less memory, e.g. I had to reduce the `TEXTURE_DIMENSION` from `3072` to `1024` in commit dc6dbfbaca7013ab6fc3b54d9b5d0db67cca9057.

Can you try the `fix_warnings` and `rosnode` branches and report back if they work for you?

**Comment by ung264** [comment]
Thanks for sharing your experience. 

When I use the fix_warnings branch, I got the same result as for master branch. I evaluate both car4-full and room4 in the format of klg and extracted RGBD images, using scripts from [dataset-tools](https://github.com/martinruenz/dataset-tools) . I obtained 5cm camera AT RMSE for car4 and 14cm camera AT RMSE for room4. They are much higher than the value indicated in the CF paper. I will open an issue about this problem in the original CF repository.
